test/user.test.js

```diff
@@ -1804,6 +1805,360 @@ describe('User', function() {
+          { 'email': updatedEmailCredentials.email,
+            'password': updatedEmailCredentials.password },
+              function(err, userInstance) {
+                if (err) return done(err);
+                assertNoAccessTokens(done);
+              });
+      });
+
+      it('invalidates sessions when email is changed using `updateOrCreate`', function(done) {
+        User.updateOrCreate(
+          { id: user.id,
+            email: updatedEmailCredentials.email,
+            password: updatedEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return done(err);
+              assertNoAccessTokens(done);
+            });
+      });
+
+      it('invalidates sessions when the email is changed using `replaceById`', function(done) {
+        User.replaceById(user.id,
+          { email: updatedEmailCredentials.email,
+            password: updatedEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return done(err);
+              assertNoAccessTokens(done);
+            });
+      });
+
+      it('invalidates sessions when the email is changed using `replaceOrCreate`', function(done) {
+        User.replaceOrCreate(
+          { id: user.id,
+            email: updatedEmailCredentials.email,
+            password: updatedEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return done(err);
+              assertNoAccessTokens(done);
+            });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `updateAttributes`', function(done) {
+        user.updateAttributes({ 'firstName': 'Janny' }, function(err, userInstance) {
+          if (err) return done(err);
+          assertUntouchedTokens(done);
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `replaceAttributes`', function(done) {
+        user.replaceAttributes(
+          { 'email': currentEmailCredentials.email,
+            'password': currentEmailCredentials.password,
+            'firstName': 'Candy' },
+            function(err, userInstance) {
+              if (err) return done(err);
+              assertUntouchedTokens(done);
+            });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `updateOrCreate`', function(done) {
+        User.updateOrCreate(
+          { id: user.id,
+            'firstName': 'Amir',
+           'email': currentEmailCredentials.email,
+           'password': currentEmailCredentials.password },
+           function(err, userInstance) {
+             if (err) return done(err);
+             assertUntouchedTokens(done);
+           });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `replaceById`', function(done) {
+        User.replaceById(user.id,
+          { 'firstName': 'Miroslav',
+            'email': currentEmailCredentials.email,
+            'password': currentEmailCredentials.password },
```

Comment:
There are more places below where your coding style does not follow ours one-arg-per-line convention, please fix them too.